ports/nrf/modules/machine/spi.c

```diff
@@ -248,75 +227,85 @@ STATIC MP_DEFINE_CONST_DICT(machine_spi_locals_dict, machine_spi_locals_dict_tab
 
 STATIC void machine_hard_spi_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     machine_hard_spi_obj_t *self = self_in;
-    spi_print(print, self->spi, false);
+    mp_printf(print, "SPI(%u)", self->p_spi.drv_inst_idx);
 }
 
 STATIC mp_obj_t machine_hard_spi_make_new(mp_arg_val_t *args) {
     // get static peripheral object
     int spi_id = spi_find(args[ARG_NEW_id].u_obj);
     const machine_hard_spi_obj_t *self = &machine_hard_spi_obj[spi_id];
 
+    nrfx_spi_config_t config;
+
     // here we would check the sck/mosi/miso pins and configure them
     if (args[ARG_NEW_sck].u_obj != MP_OBJ_NULL
         && args[ARG_NEW_mosi].u_obj != MP_OBJ_NULL
         && args[ARG_NEW_miso].u_obj != MP_OBJ_NULL) {
 
-        self->spi->init.clk_pin = args[ARG_NEW_sck].u_obj;
-        self->spi->init.mosi_pin = args[ARG_NEW_mosi].u_obj;
-        self->spi->init.miso_pin = args[ARG_NEW_miso].u_obj;
+        config.sck_pin  = ((const pin_obj_t *)args[ARG_NEW_sck].u_obj)->pin;
+        config.mosi_pin = ((const pin_obj_t *)args[ARG_NEW_mosi].u_obj)->pin;
+        config.miso_pin = ((const pin_obj_t *)args[ARG_NEW_miso].u_obj)->pin;
     } else {
-        self->spi->init.clk_pin = &MICROPY_HW_SPI0_SCK;
-        self->spi->init.mosi_pin = &MICROPY_HW_SPI0_MOSI;
-        self->spi->init.miso_pin = &MICROPY_HW_SPI0_MISO;
+        config.sck_pin  = (&MICROPY_HW_SPI0_SCK)->pin;
+        config.mosi_pin = (&MICROPY_HW_SPI0_MOSI)->pin;
+        config.miso_pin = (&MICROPY_HW_SPI0_MISO)->pin;
     }
 
+    // Manually trigger slave select from upper layer.
+    config.ss_pin = NRFX_SPI_PIN_NOT_USED;
+
     int baudrate = args[ARG_NEW_baudrate].u_int;
 
     if (baudrate <= 125000) {
-        self->spi->init.freq = HAL_SPI_FREQ_125_Kbps;
+        config.frequency = NRF_SPI_FREQ_125K;
```

Comment:
I *think* we can make this code a lot smaller by calculating the frequency from the baudrate directly (something like `config.frequency = baudrate * 268`). But that can be done after this PR is merged, as it is not directly related to the nrfx changes.